media-library-client/src/app/private-container/pages/home/video-gallery/video-tiles/video-player/video-player.tsx

```diff
@@ -0,0 +1,165 @@
+import Container from '@mui/material/Container';
+import { SyntheticEvent, useRef, useState } from 'react';
+import ReactPlayer from 'react-player';
+import screenfull from 'screenfull';
+import ControlIcons from './video-player-controls';
+import { ReactPlayerProps } from 'react-player';
+import './video-player.scss'
+import VideoPlayerProps from './video-player-props';
+
+const format = (seconds: number) => {
+    if (isNaN(seconds) || seconds < 0) {
+      return '00:00';
+    }
+
+    const date = new Date(seconds * 1000);
+    const hh = date.getUTCHours();
+    const mm = date.getUTCMinutes();
+    const ss = date.getUTCSeconds().toString().padStart(2, "0");
+
+    if (hh) {
+        return `${hh}:${mm.toString().padStart(2, "0")}:${ss}`
+    } else {
+        return `${mm}:${ss}`
+    }
+};
+
+export default function VideoPlayer({URL}: Readonly<VideoPlayerProps>) {
+    const [playerstate, setPlayerState] = useState({
+        playing: true,
+        muted: false,
+        volume: 0.5,
+        playerbackRate: 1.0,
+        played: 0,
+        seeking: false,
+    })
+
+
+    //Destructure State in other to get the values in it
+    const { playing, muted, volume, playerbackRate, played } = playerstate;
+    const playerRef = useRef<ReactPlayer | null>(null);
+    const playerDivRef = useRef(null);

```

Comment:
Should this value be initialised as "false" rather than "null" instead considering its use. Also is the "useRef"  correctly triggering the redraws as needed? If not a "useState" + "useEffect" would be better.